pool.go

```diff
@@ -0,0 +1,63 @@
+package Go_Pool
+
+import (
+	"errors"
+	"runtime"
+	"sync"
+)
+
+type Pool struct {
+	Tasks       []*Task
+	concurrency int
+	numCPUs     int
+	tasksChan   chan *Task
+	wg          sync.WaitGroup
+}
+
+var pool *Pool
+var once sync.Once
+
+func NewPool(tasks []*Task, concurrency int, numCPUs int) *Pool {
```

Comment:
- Please make sure multiple clients will be able to send tasks after pool server starts.

Reply:
Done